File: ddz_server/src/main/java/com/zyd/ddz/constant/CharacterType.java
package com.zyd.ddz.constant;

/**
 * @author zyd
 * @date 2023/4/11 13:30
 */
public enum CharacterType {
    /**
     * 地主
     */
    LANDOWNER,
    /**
     * 农民
     */
    FARMER,
    /**
     * 搭档
     */
    PARTNER;

    public static CharacterType getType(int type){
        return values()[type - 1];
    }
}

File: client/assets/scripts/constant/CharacterType.ts
import { _decorator, Component, Node } from 'cc';
const { ccclass, property } = _decorator;

export enum CharacterType {
    LANDOWNER = 1,

    FARMER = 2,

    PARTNER = 3,

    
}

export class Role {
    public static SELF = 1;
    public static SIDE = 2;
}


